host/src/lib.rs

```diff
@@ -134,13 +135,12 @@ impl From<&Opts> for TaskManagerOpts {
     }
 }
 
-pub type TaskChannelOpts = (ProofRequest, Opts, SupportedChainSpecs);
-
 #[derive(Debug, Clone)]
 pub struct ProverState {
     pub opts: Opts,
     pub chain_specs: SupportedChainSpecs,
-    pub task_channel: mpsc::Sender<TaskChannelOpts>,
+    pub task_channel: mpsc::Sender<ProofRequest>,
+    pub cancel_channel: mpsc::Sender<TaskDescriptor>,
```

Comment:
Just wondering if we can make update/query also channel signal, even maybe a all-in-one request handle channel, then we don't need to pass Arc<TaskManager> everywhere? Problems could be hard to abstract each request (in 1 API) as so many different types. Not sure if it's worth.

Reply:
The cancel channel is handled outside of handling the task channel because it needs to be able to cancel tasks that are currently being handled, meaning it has to be done in parallel

Reply:
Yes, I just thought if we use a unified channel (in my mind it's pretty much like golang channel, maybe wrong), that in parallel operation is inside proverState only, and provers does not need to get taskdb from proverState to query things & use different channels to operate different tasks. it seems there are advantages, is it?

Reply:
I understand what you were talking about now, have a look at the updated implementation

Reply:
not a strong opt, just for design discussion, if we both think it's worth, we can refine it later.